inc/datahandlers/user.php

```diff
@@ -888,6 +888,20 @@ function verify_checkfields()
 	}
 
 	/**
+	 * Verifies if the selected timezone is valid.
+	 *
+	 * @return boolean True when valid, false when invalid.
+	 */
+	function verify_timezone()
+	{
+		$user = &$this->data;
+
+		$timezones = get_supported_timezones();
+
+		return array_key_exists($user['timezone'], $timezones);
```

Comment:
> Why would you force the default value if it is wrong?

Because we do it for other options too?
https://github.com/mybb/mybb/blob/feature/inc/datahandlers/user.php#L638-654
https://github.com/mybb/mybb/blob/feature/inc/datahandlers/user.php#L671
https://github.com/mybb/mybb/blob/feature/inc/datahandlers/user.php#L681
etc.